### ISSUE TITLE
no-reserved-keywords not finding "catch" in all places

### ISSUE THREAD

**Issue by chrismbarr** [palantir/tslint#3311]
### Bug Report

- __TSLint version__: 5.7.0
- __TypeScript version__: 2.5.3
- __Running TSLint via__:  gulp

#### TypeScript code being linted

```ts
spyOn(customerService, "getCustomer").and.returnValue({
  then: (): {catch: () => void} => {
    return {
      catch: (): void => { return; }
    };
  }
});
```

with `tslint.json` configuration:

```json
"no-reserved-keywords": true
```

#### Actual behavior

It only reports a violation in the typedef

![image](https://user-images.githubusercontent.com/463685/31462378-94a0e968-ae9a-11e7-85f1-44ad8f7e23b7.png)

#### Expected behavior
I would expect it to also report a violation in the return statement below where the actual object property is named `catch`, in addition to the typedef above.

**Comment by ajafff** [comment]
There is no TSLint core rule called `no-reserved-keywords`. This must be a custom rule.

**Comment by chrismbarr** [comment]
Oh wow, terribly sorry.  It's a [rule from microsoft](https://github.com/Microsoft/tslint-microsoft-contrib)!
